packages/devtools_app/lib/src/profiler/cpu_profile_controller.dart

```diff
@@ -51,6 +55,35 @@ class CpuProfilerController with SearchControllerMixin<CpuStackFrame> {
 
   Iterable<String> get userTags => _dataByTag[userTagNone]?.userTags ?? [];
 
+  /// The toggle filters available for the CPU profiler.
+  ///
+  /// This is a getter so that `serviceManager` is initialized by the time we
+  /// access this list.
+  // TODO(kenz): make this lazy once we migrate to null safety.
+  List<ToggleFilter> get toggleFilters => _toggleFilters ??= [
+        ToggleFilter<CpuStackFrame>(
+          name: 'Hide Native code',
+          includeCallback: (stackFrame) => stackFrame.url.isNotEmpty,
+        ),
+        ToggleFilter<CpuStackFrame>(
+          name: 'Hide core Dart libraries',
+          includeCallback: (stackFrame) =>
+              !stackFrame.url.contains('org-dartlang-sdk'),
+        ),
+        if (serviceManager.connectedApp?.isFlutterAppNow ?? true)
+          // TODO(kenz): should we also hide flutter engine frames as part of this
+          // filter? We don't have a url to go off of but we could check for a
+          // stack frame name containing 'flutter::'
+          ToggleFilter<CpuStackFrame>(
+            name: 'Hide core Flutter libraries',
+            includeCallback: (stackFrame) =>
+                !(stackFrame.url.contains('packages/flutter/') ||
```

Comment:
should be startsWith not contains to avoid false positives. Also add the engine as users really really don't care about the engine.

Reply:
we can't use starts with because the beginning of the url will change depending on which flutter a user is using. For example, file:///path/to/my/flutter/checkout/packages/flutter...

Reply:
hmm. what is we pre-process all stackFrame urls to use our best guess at package: and dart: before they get to this point?
that way the check can be `.startsWith('package:flutter/')`

Reply:
done

Reply:
How would you suggest filtering out flutter engine code? We don't have a uri for flutter engine stack frames since. Should we check for stack frame names that start with `flutter::`? Related, what about stack frame names that start with `dart::` - what category do those stack frames belong to?